testsuites/basic/lsvmbus.py

```diff
@@ -0,0 +1,112 @@
+from lisa.tools import Lscpu, Lsvmbus
+from lisa.util import LisaException
+from lisa.util.shell import wait_tcp_port_ready
+
+
+@TestSuiteMetadata(
+    area="core",
+    category="functional",
+    description="""
+    This test suite uses to check correctness of vmbus channel
+    """,
+    tags=[],
+)
+class LsVmBus(TestSuite):
+    TIME_OUT = 300
+    __Tokens = [
+        "Operating system shutdown",
+        "Time Synchronization",
+        "Heartbeat",
+        "Data Exchange",
+        "mouse",
+        "keyboard",
+        "Synthetic network adapter",
+        "Synthetic SCSI Controller",
+    ]
+
+    @TestCaseMetadata(
+        description="""
+        This test is to check channel count of each netvsc and storvsc SCSI device.
+        When count is unexpected, then test fail.
+        """,
+        priority=0,
+        requirement=simple_requirement(
+            environment_status=EnvironmentStatus.Deployed,
+            supported_features=[SerialConsole],
+        ),
+    )
+    def lsvmbus_test(self, case_name: str) -> None:
+        node = self.environment.default_node
+        case_path: Optional[Path] = None
+
+        is_ready, tcp_error_code = wait_tcp_port_ready(
+            node.public_address, node.public_port, log=self.log, timeout=self.TIME_OUT
+        )
+        if not is_ready:
+            serial_console = node.features[SerialConsole]
+            case_path = self._create_case_log_path(case_name)
+            serial_console.check_panic(saved_path=case_path, stage="bootup")
+            raise LisaException(
+                f"cannot connect to [{node.public_address}:{node.public_port}], "
+                f"error code: {tcp_error_code}, no panic found in serial log"
+            )
+        node.is_connected = True
+        environment_information = node.get_node_information()
+        if environment_information["vm_generation"] == "1":
+            self.__Tokens.append("Synthetic IDE Controller")
+
+        lsvmbus_tool = node.tools[Lsvmbus]
+        vmbus_results = lsvmbus_tool.get_verbose_messages()
+        for token in self.__Tokens:
+            res = [key for key, val in vmbus_results.items() if token in key]
+            if res and len(res) > 0:
+                self.log.info(
+                    f"{token} is in output of lsvmbus, matched times {len(res)}"
+                )
+            else:
+                raise LisaException(f"{token} is not in output of lsvmbus")
+
+        # get actual core count
+        lscpu_tool = node.tools[Lscpu]
+        core_count = lscpu_tool.get_core_count()
+        # Each netvsc device should have "the_number_of_vCPUs" channel(s)
+        # with a cap value of 8.
+        expected_network_channel_count = core_count
+        if core_count > 8:
+            expected_network_channel_count = 8
+
+        # Each storvsc SCSI device should have "the_number_of_vCPUs / 4" channel(s)
+        # with a cap value of 64.
+        expected_scsi_channel_count = math.ceil(core_count / 4)
```

Comment:
similar as above, and simplify used numbers to help clearer understanding.
try `math.ceil(min(core_count, 256)/4)`